### PR TITLE
Development

### PR DESCRIPTION
**Pull request by olanrewaju139**

### CODE REVIEW

File: CMP1903_A1_2324/Die.cs
Comment (black-eye-inc) [inline CMP1903_A1_2324/Die.cs:1]: "public int HoldValue { get; private set; }":
I'd recommend using _camelCase (calling it "_holdValue") for private parameters (as instructed by "C# style guidelines")
one of the Object-Oriented Principles is encapsulation (making all parameters (within reason) private).

File: CMP1903_A1_2324/Game.cs
Comment (black-eye-inc) [inline CMP1903_A1_2324/Game.cs:1]: I'd recommend also adding EXTRA functionality "continuous roll method" -- this is basically a function that runs your "Start()" method and then asks the user if they'd like to roll again or exit the function.

File: CMP1903_A1_2324/Program.cs
Comment (black-eye-inc) [inline CMP1903_A1_2324/Program.cs:1]: **Summary of my entire review:**
Your program works great and error free

All commenting is great -- neither over explaining or under explaining how the code works.

The only problems sighted are:
- The “Debug.Assert()” statement in the “Testing.GameTest” needs working on so it trigger propperly
-  Optional: can implement the extra functionality Continual Roll method in “game” class

Keep up the good work and I have full faith you'll pass Computer Science!

C# Style Guide:
- Overall a great compliance with the C# style guide's naming rules (e.g: using PascalCase for all method names), with only one slip-up with "HoldValue" in "Die.cs".
- Overall Good compliance with all the C# Style guide rules.
- Here's the URL for the  C# Style Guide encase you want it: https://google.github.io/styleguide/csharp-style.html

XML Documentation:
- No have found no attempt at XML Documentation in the program
- Encase you want to implement it, here's is a good URL on how implement to XML Documentation: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments

File: CMP1903_A1_2324/Testing.cs
Comment (black-eye-inc) [inline CMP1903_A1_2324/Testing.cs:1]: for line 21 " Debug.Assert(test > 18 || test>=3,"Game test unsuccessful")":
Please explain your use of "Debug.Assert(...)" -- "Debug.Assert(...)" only flags an error if the input argument (in this case "test > 18 || test>=3") is "false".
The input here ("test > 18 || test>=3") will always return "true" if test >= 3, I'd recommend "test >= 3 && test < 19" (keeping it in the same style as the next "Debug.Assert(...)" you use)


Line 33 "Debug.Assert(outcome >= 1 && outcome < 7, "Die test unsuccessful");":
Flawless implementation of "Debug.Assert(...)" here

File: CMP1903_A1_2324/Die.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿using System;
```
Comment (norbertchidubem15) [inline CMP1903_A1_2324/Die.cs:1]: Your method naming with GameTest effectively communicates its purpose..Impressive.

**Review by black-eye-inc — CHANGES_REQUESTED** [review]
the implementation of "Debug.Assert()" statement in "Testing.cs" will need to be modified for the program to meets all required criteria mentioned in the brief.

I'd recommend creating the EXTRA Functionality requirement of being able to "continuously roll the dice" in "Game.cs"

File: CMP1903_A1_2324/Program.cs
Comment (norbertchidubem15) [reply CMP1903_A1_2324/Program.cs:1]: Overall, the Game class is off to a good start! It efficiently handles the core functionality of rolling dice and calculating totals. Nice keep up the good work!

File: CMP1903_A1_2324/Testing.cs
Comment (norbertchidubem15) [inline CMP1903_A1_2324/Testing.cs:1]: Keep it up you did a nice job!!

**Review by norbertchidubem15 — COMMENTED** [review]
Looks great you did a nice job....

File: CMP1903_A1_2324/Die.cs [hunk 2]
```diff
@@ -8,15 +8,22 @@ namespace CMP1903_A1_2324
...
-         * and one method that rolls the die, returns and integer and takes no parameters.
-         */
-
-        //Property
-
+        
 
         //Method
+        public int holdValue { get; private set; } //to store the value gotten from rolling the dice
+        // <black-eye-inc> Code Review:
```
Comment (black-eye-inc) [inline CMP1903_A1_2324/Die.cs:15]: I'd recommend using _camelCase (calling it "_holdValue") for private parameters (as instructed by "C# style guidelines")
One of the Object-Oriented Principles is encapsulation (making all parameters (within reason) private).

File: CMP1903_A1_2324/Testing.cs [hunk 3]
```diff
@@ -8,13 +9,37 @@ namespace CMP1903_A1_2324
...
+        {
+            //creates a game object
+            Game game = new Game();
+            Console.WriteLine("TestGame");
+            int test = game.Start(); //this uses the game class method
 
-        //Method
+
+            Debug.Assert(test > 18 || test>=3,"Game test unsuccessful"); // compares the outcome of the game and check if the total is
+            //between 3-18, if otherwise the assertation catches it
```
Comment (black-eye-inc) [inline CMP1903_A1_2324/Testing.cs:22]: The condition in the Debug.Assert allows any number bigger than 3 to enter
I'd personality change from "test > 18 || test>=3" to "test <= 18 || test>=3"

File: CMP1903_A1_2324/Testing.cs [hunk 4]
```diff
@@ -8,13 +9,37 @@ namespace CMP1903_A1_2324
...
+            // Please explain your use of "Debug.Assert(...)" -- "Debug.Assert(...)" only flags an error if the input argument
+            // (in this case "test > 18 || test>=3") is "false"
+            // the input here "test > 18 || test>=3" will always return "true" if test >= 3, I'd recommend "test >= 3 && test < 19"
+            // (keeping it in the same style as the next "Debug.Assert(...)" you use)
+            // </black-eye-inc>
+
+            Die die = new Die();
+            int outcome = die.Roll(); //creates a die object
+            
+            Debug.Assert(outcome >= 1 && outcome < 7, "Die test unsuccessful");// compares the output of the die
```
Comment (black-eye-inc) [inline CMP1903_A1_2324/Testing.cs:33]: Flawless Debug.Assert here - good job!

File: CMP1903_A1_2324/Game.cs [hunk 5]
```diff
@@ -8,14 +8,34 @@ namespace CMP1903_A1_2324
...
+
+                Console.WriteLine($"die {i + 1} is: {roll}"); // Print the output of each roll
+            }
+
+            Console.WriteLine("total is: " + total); // Print the total
+
+            return total;// returns the sum of the rolls
+
+        }
+        // <black-eye-inc>
```
Comment (black-eye-inc) [inline CMP1903_A1_2324/Game.cs:34]: You could add the extra functionality "Continuous roll" method for higher marks

**Review by black-eye-inc — COMMENTED** [review]
Overall simple and effective job!
There is only one problematic line I've found (that being the Debug Assert Line) but once that's done the code meets all basic criteria!
Keep up the good work!

**Pull request merged**